### PR TITLE
Improve initial options

### PR DESCRIPTION
**Pull request by lightsprint09**
Use predefined keys in the dictionary to prevent typos

My initial intention was to remove key-value at all.
The initial options should be type save as well, like

ForceWebsocket(<expect Bool>)
CustomLogger(<expect Logger Implementation>)

But after all the solution should be easy to transfer to an [String: AnyObject]. Any ideas?

### CODE REVIEW

**Comment by nuclearace** [comment]
I don't know if this a good approach. I don't think that typos in options is really that big of a problem.

**Comment by lightsprint09** [comment]
I think it helps. And does not break anything

But a real type save solution would be even better

**Comment by lightsprint09** [comment]
this commit partly proves it https://github.com/socketio/socket.io-client-swift/commit/5d867f4f6524f302d52fdc60c6907b6687dc8733 ;)

look at AbstractTest

**Comment by nuclearace** [comment]
Heh, yeah. I've got another solution in mind though

**Comment by lightsprint09** [comment]
Ok? What other solution?

**Comment by nuclearace** [comment]
Similar to what you had, but using an enum.
```swift
public typealias SocketOptionsDictionary = [SocketIOClientOptions: AnyObject]

public enum SocketIOClientOptions: String {
    case ConnectParams = "connectParams"
    case Reconnects = "reconnect"
    case ReconnectAttempts = "reconnectAttempts"
    case ReconnectWait = "reconnectWait"
    case ForcePolling = "forcePolling"
    case ForceWebsockets = "forceWebsockets"
    case Nsp = "nsp"
    case Cookies = "cookies"
    case Log = "log"
    case Logger = "logger"
    case SessionDelegate = "sessionDelegate"
    case Path = "path"
    case ExtraHeaders = "extraHeaders"
    case HandleQueue = "handleQueue"
}
```

Would need a convenience init on the client for Objective-C

**Comment by lightsprint09** [comment]
Like this?

```swift
enum SocketInitialOptionsEnum {
    case ConnectionParameter([String: AnyObject]?)
    case Reconnects(Bool)

    func toString() -> String {
        switch self {
        case ConnectionParameter:
            return "connectParams"
        case Reconnects:
            return "reconnects"
        }
    }
}
```

**Comment by nuclearace** [comment]
I was thinking something more along the lines of https://github.com/socketio/socket.io-client-swift/commit/40ac53844d908e65cfb870ca27cfa2356c3034e1

**Comment by nuclearace** [comment]
That way also has the benefit in that it won't be a compatibility breaking change.

File: SocketIOClientSwift/SocketInitialOptions.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//  InitialOptions.swift
+//  Socket.IO-Client-Swift
+//
+//  Created by Lukas Schmidt on 17.10.15.
+//
+//
+
+import Foundation
+
+struct SocketInitialOptions {
```
Comment (NachoSoto) [inline SocketIOClientSwift/SocketInitialOptions.swift]: If you make this an `enum` with associated `String` values instead, you can statically guarantee that all keys are unique.

**Comment by nuclearace** [comment]
#212 Is probably the better way to go about it.

**Pull request closed without merging**